actionview/lib/action_view/helpers/capture_helper.rb

```diff
@@ -40,9 +40,16 @@ module CaptureHelper
       #
       #   @greeting # => "Welcome to my shiny new web page! The date and time is 2018-09-06 11:09:16 -0500"
       #
-      def capture(*args)
+      def capture(*args, &block)
         value = nil
-        buffer = with_output_buffer { value = yield(*args) }
+        block_context = block.binding.receiver
+
+        buffer = if block_context.respond_to?(:render_in) && block_context.respond_to?(:with_output_buffer)
```

Comment:
I'd love thoughts here, maybe it would be useful to raise a helpful error message here if `block_context` responds to `render_in` but not `with_output_buffer` since we'll fall back to the incorrect behavior.